### PR TITLE
Authn delete chain

### PR DESCRIPTION
**Pull request by savonarola**

### CODE REVIEW

File: apps/emqx/src/emqx_authentication_config.erl [hunk 1]
```diff
@@ -52,7 +52,6 @@
 -type update_request() ::
     {create_authenticator, chain_name(), map()}
     | {delete_authenticator, chain_name(), authenticator_id()}
-    | {delete_authenticators, chain_name()}
```
Comment (thalesmg) [inline apps/emqx/src/emqx_authentication_config.erl:55]: Just confirming: after this change, the user will have to delete each authenticator individually to clear the chain?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx/src/emqx_authentication_config.erl (hunk 1 above)
Comment (savonarola) [reply apps/emqx/src/emqx_authentication_config.erl:55]: Yes. After all authenticators are removed, the chain is removed too.

Comment (thalesmg) [reply apps/emqx/src/emqx_authentication_config.erl:55]: Just curiosity: any particular reason to remove this API?

File: apps/emqx/src/emqx_authentication.erl [hunk 2]
```diff
@@ -444,50 +445,36 @@ handle_call(
...
-            true = ets:delete(?CHAINS_TAB, Name),
-            reply(ok, maybe_unhook(State))
-    end;
+handle_call({delete_chain, ChainName}, _From, State) ->
+    UpdateFun = fun(Chain) ->
+        {_MatchedIDs, NewChain} = do_delete_authenticators(fun(_) -> true end, Chain),
+        {ok, ok, NewChain}
+    end,
+    Reply = with_chain(ChainName, UpdateFun),
+    reply(Reply, maybe_hook(State));
```
Comment (thalesmg) [inline apps/emqx/src/emqx_authentication.erl:454]: It was `maybe_unhook` [before](https://github.com/emqx/emqx/pull/8108/files#diff-9d800c0108be2295c2e91d3fbfcf060b13d5c6ede0b3a544751e5021a4cee0b4L466).  Is the change to `maybe_hook` intentional?

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx/src/emqx_authentication.erl (hunk 2 above)
Comment (savonarola) [reply apps/emqx/src/emqx_authentication.erl]: ✅

**Review by thalesmg — DISMISSED** [review]

**Review by HJianBo — DISMISSED** [review]

**Review by thalesmg — APPROVED** [review]

**Pull request merged**